compiler/rustc_codegen_llvm/src/back/target_machine_wrapper.rs

```diff
@@ -0,0 +1,98 @@
+use std::{
+    ffi::{c_char, CStr},
+    marker::PhantomData,
+    ops::Deref,
+    ptr::NonNull,
+};
+
+use rustc_data_structures::small_c_str::SmallCStr;
+
+use crate::{errors::LlvmError, llvm};
+
+#[repr(transparent)]
+pub struct TargetMachineWrapper {
+    tm: NonNull<llvm::TargetMachine>,
+    phantom: PhantomData<&'static mut llvm::TargetMachine>,
+}
+
+impl TargetMachineWrapper {
+    pub fn new(
+        triple: &CStr,
```

Comment:
we take all kinds of borrows here as arguments, but out result is owned. is this correct, does `LLVMRustCreateTargetMachine` create copies of these strings or just use them for some lookup or does it borrow from the input? I suspect that it doesn't borrow but it would be nice to confirm that and write a comment about that on the `unsafe` block below.

Reply:
Checked it again, TargetMachine internally generally std::string which makes copies. There are only two expections, but those are handled by allocating on the heap and freeing it when we dispose of target machine.

Added a comment to the unsafe block.